scrape/scrape.go

```diff
@@ -1484,32 +1484,49 @@ func (sl *scrapeLoop) append(app storage.Appender, b []byte, contentType string,
 		sl.cache.iterDone(len(b) > 0)
 	}()
 
+	var skipCallNextEntry bool
+	var foundEOF bool
+	var et textparse.Entry
+	var trimedMetric []byte
 loop:
 	for {
+		// When parsing created timestamps from OpenMetrics text format,
+		// it is possible that we find EOF early. Therefore we break the loop.
+		if foundEOF {
+			break
+		}
 		var (
-			et                       textparse.Entry
 			sampleAdded, isHistogram bool
 			met                      []byte
 			parsedTimestamp          *int64
 			val                      float64
+			typ                      model.MetricType
+			help                     []byte
+			unit                     []byte
 			h                        *histogram.Histogram
 			fh                       *histogram.FloatHistogram
 		)
-		if et, err = p.Next(); err != nil {
-			if errors.Is(err, io.EOF) {
-				err = nil
+		if !skipCallNextEntry {
```

Comment:
Wondering if it would be simpler to re-arrange so `Next` has always been called at the start of the loop?

Reply:
I can't see how that could be done... (or maybe I couldn't understand what you mean)

For OM-text, we need to parse 2 lines to identify created timestamps and the parser can't go back once a line is parsed, therefore we need to skip parsing yet another line if we haven't found a _created line in the last call 🤔